remote_engine_client/src/client.rs

```diff
@@ -532,6 +543,48 @@ impl Stream for ClientReadRecordBatchStream {
                     }.fail()));
                 }
 
+                match response.metrics {
```

Comment:
As discussing in following, metric stream can be called before all data stream are polled to end which led to the polling ends by mistake(end while finding output is None).